### PR TITLE
docs(readme): improve typescript documentation

### PR DESCRIPTION
**Pull request by toddbluhm**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Reorganized TypeScript/Flow docs to place TypeScript first and added an example for setting up `ts-standard`. Changes based on the comment here: https://github.com/standard/standard/pull/1595#issuecomment-725026148

**Which issue (if any) does this pull request address?**
https://github.com/standard/standard/issues/1590

**Is there anything you'd like reviewers to focus on?**
Language is appropriate and accurate and the example looks simple enough.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -507,10 +507,46 @@ Or, add this to `package.json`:
...
+[`ts-standard`](https://github.com/standard/ts-standard) is the officially supported variant for
+TypeScript. `ts-standard` supports all the same rules and options as `standard` and includes
+additional TypeScript specific rules. `ts-standard` will even lint regular `javascript` files
+by setting the configuration in `tsconfig.json`.
+
+```bash
+npm install ts-standard --save-dev
+```
+
+Then run (where `tsconfig.json` is located in the working director):
```
Comment (munierujp) [inline README.md:528]: Fix typo.

```suggestion
Then run (where `tsconfig.json` is located in the working directory):
```

**Review by Divlo — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -507,10 +507,46 @@ Or, add this to `package.json`:
 `standard` supports the latest ECMAScript features. However, Flow and TypeScript add new
 syntax to the language, so they are not supported out-of-the-box.
 
-To support JavaScript language variants, `standard` supports specifying a custom JavaScript
+For Typescript, an official variant `ts-standard` is supported and maintained that provides a very
```
Comment (munierujp) [inline README.md:510]: Fix typo

```suggestion
For TypeScript, an official variant `ts-standard` is supported and maintained that provides a very
```

**Review by munierujp — CHANGES_REQUESTED** [review]
see my comments :bow:

File: README.md (hunk 2 above)
Comment (toddbluhm) [reply README.md]: Done

**Review by munierujp — COMMENTED** [review]
LGTM 😄

**Review by munierujp — APPROVED** [review]

**Pull request merged**